static/js/shared/data_fetcher.ts

```diff
@@ -387,108 +146,112 @@ function computePerCapita(
-  let dcidParams = `?`;
+  let placeParams = `?`;
   for (const place of places) {
-    dcidParams += `&dcid=${place}`;
+    placeParams += `&dcid=${place}`;
   }
-  const apiDataPromises: Promise<AxiosResponse<StatApiResponse>>[] = [];
-  for (const statsVar of statsVars) {
-    apiDataPromises.push(axios.get(`/api/stats/${statsVar}${dcidParams}`));
-  }
-  if (perCapita) {
-    apiDataPromises.push(
-      axios.get(`/api/stats/${TOTAL_POPULATION_SV}${dcidParams}`)
-    );
-  } else {
-    for (const denom of denominators) {
-      apiDataPromises.push(axios.get(`/api/stats/${denom}${dcidParams}`));
-    }
-  }
-  const displayNamesPromise: Promise<AxiosResponse<
-    DisplayNameApiResponse
-  >> = axios.get(`/api/place/displayname${dcidParams}`);
+  const displayNamesPromise: Promise<DisplayNameApiResponse> = axios
+    .get(`/api/place/displayname${placeParams}`)
+    .then((resp) => {
+      return resp.data;
+    });
 
   // create list of promises containing apiDataPromises followed by displayNamesPromise
-  const apiPromises: Promise<AxiosResponse<any>>[] = [];
-  apiPromises.push(...apiDataPromises);
-  apiPromises.push(displayNamesPromise);
+  const apiPromises: Promise<StatApiResponse | DisplayNameApiResponse>[] = [
+    statDataPromise,
+    denomDataPromise,
+    displayNamesPromise,
+  ];
 
   return Promise.all(apiPromises).then((allResp) => {
-    const result = new StatsData(places, statsVars, [], {}, "");
+    const result = new StatData(places, statVars, [], {}, "");
     const numOccurencesPerDate: { [key: string]: number } = {};
     const numStatVarsPerPlace: { [key: string]: number } = {};
-    const displayNameMapping =
-      allResp.length > 0 && allResp[allResp.length - 1].data
-        ? allResp[allResp.length - 1].data
-        : {};
-    for (let i = 0; i < numStatsVars; i++) {
-      const sv = statsVars[i];
-      result.data[sv] = allResp[i].data;
-      if (perCapita) {
-        result.data[sv] = computePerCapita(
-          allResp[i].data,
-          allResp[numStatsVars].data,
-          scaling
-        );
-      } else if (denominators.length) {
-        result.data[sv] = computePerCapita(
-          allResp[i].data,
-          allResp[i + numStatsVars].data,
-          scaling
-        );
+    const statResp = allResp[0] as StatApiResponse;
+    const denomResp = allResp[1] as StatApiResponse;
+    const displayNameMapping = (allResp[2]
+      ? allResp[2]
+      : {}) as DisplayNameApiResponse;
+
+    for (const place in statResp) {
+      const placeData = statResp[place].data;
+      for (const statVar in placeData) {
+        if (perCapita) {
+          placeData[statVar] = computePerCapita(
+            placeData[statVar],
+            denomResp[place].data[TOTAL_POPULATION_SV],
+            scaling
+          );
+        } else if (statVar in denominators) {
+          placeData[statVar] = computePerCapita(
+            placeData[statVar],
+            denomResp[place].data[denominators[statVar]],
```

Comment:
is it guaranteed that place will be in denomResp? If not, probably want to add a check

Reply:
Yeah, the API will always return entries for given places, ex: https://api.datacommons.org/v1/stat/set/series?places=dummy&stat_vars=Count_Person